### PR TITLE
WIP: Remove inconsistencies in scipy.misc / scipy.ndimage

### LINKED ISSUE #6242: Inconsistency / duplication for imread and imshow, imsave
## The issue
- [`imread`](http://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.imread.html#scipy-ndimage-imread) is in scipy.ndimage and [`scipy.misc.imread`](http://docs.scipy.org/doc/scipy/reference/generated/scipy.misc.imread.html)
- [`imshow`](http://docs.scipy.org/doc/scipy/reference/generated/scipy.misc.imshow.html) and [`imsave`](http://docs.scipy.org/doc/scipy/reference/generated/scipy.misc.imsave.html) are in scipy.misc

This seems inconsistent / unlogical from a users perspective.

## Proposed solution

Moving some to [`scipy.ndimage`](http://docs.scipy.org/doc/scipy/reference/ndimage.html#utility) and others to [`scipy.io`](https://docs.scipy.org/doc/scipy/reference/io.html):
- [ ] `scipy.misc.imshow` → `scipy.ndimage.imshow`
- [ ] `scipy.ndimage.imread`, `scipy.misc.imread` → `scipy.io.imread`
- [ ] `scipy.misc.imsave` → `scipy.io.imwrite`
- [x] `scipy.misc.imfilter` → `scipy.ndimage.imfilter`
- [x] `scipy.misc.imresize` → `scipy.ndimage.imresize`
- [x] `scipy.misc.imrotate` → `scipy.ndimage.imrotate`
- [ ] `scipy.misc.ascent` → `scipy.ndimage.ascent`
- [ ] `scipy.misc.face` → `scipy.ndimage.face`

I'm not sure about the remaining `bytescale`, `fromimage`, `toimage`.

## What needs to be done
- Moving the code from `pilutil.py` to `io.py` (and importing the functions from `io.py` in `pilutil.py`)
- Adding deprecation warnings to `pilutil.py`
- Adjusting tests
- Adjusting documentation

### PR DESCRIPTION
**Pull request by MartinThoma**
* Move scipy.misc.imfilter to scipy.ndimage.imfilter
* Move scipy.misc.imresize to scipy.ndimage.imresize
* Move scipy.misc.imrotate to scipy.ndimage.imrotate

Fixes #6242

### CODE REVIEW

**Comment by MartinThoma** [comment]
There is still quite a bit to be done for this one. However, I wanted to make this pull request to run the tests (when I execute `runtests.py` on my machine I get a lot of errors which seem to be unrelated).

**Comment by MartinThoma** [comment]
Could somebody please explain why importing `toimage` does not work anymore?

```
======================================================================
ERROR: test_pilutil.test_imread_4bit
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/local/lib/python2.7/dist-packages/numpy/testing/decorators.py", line 147, in skipper_func
    return f(*args, **kwargs)
  File "/home/moose/GitHub/scipy/build/testenv/lib/python2.7/site-packages/scipy/misc/tests/test_pilutil.py", line 201, in test_imread_4bit
    im = misc.imread(f)
AttributeError: 'module' object has no attribute 'imread'

======================================================================
ERROR: Failure: ImportError (cannot import name toimage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/moose/GitHub/scipy/build/testenv/lib/python2.7/site-packages/scipy/ndimage/__init__.py", line 161, in <module>
    from .filters import *
  File "/home/moose/GitHub/scipy/build/testenv/lib/python2.7/site-packages/scipy/ndimage/filters.py", line 37, in <module>
    from scipy.misc import doccer, toimage, fromimage
ImportError: cannot import name toimage
```

File: scipy/misc/pilutil.py [hunk 1]
```diff
@@ -28,6 +27,14 @@
 __all__ = ['fromimage', 'toimage', 'imsave', 'imread', 'bytescale',
            'imrotate', 'imresize', 'imshow', 'imfilter']
 
+from ndimage import imresize, imfilter, imrotate
```
Comment (rgommers) [inline scipy/misc/pilutil.py]: This doesn't look right, should be ``from scipy.ndimage ...``.

File: scipy/misc/pilutil.py [hunk 2]
```diff
@@ -28,6 +27,14 @@
...
+imresize = numpy.deprecate(imresize,
+                           message='misc.imresize is deprecated in Scipy 1.0')
```
Comment (rgommers) [inline scipy/misc/pilutil.py]: Can you add that the function was moved to ``ndimage`` in the message?

Comment (rgommers) [inline scipy/misc/pilutil.py]: gh-6242 had some discussion on whether these functions shouldn't just be deprecated unconditionally (https://github.com/scipy/scipy/issues/6242#issuecomment-225426614). There's something to say for that. Good to get a decision on that first.

**Comment by rgommers** [comment]
> Could somebody please explain why importing toimage does not work anymore?

That'll be fixed if you correct the ``ndimage`` import I just pointed out.

**Comment by rgommers** [comment]
I suggest to work on the move of ``imwrite`` and ``imsave`` to ``io`` first, because those are the only two functions that we clearly want to keep. 

Need some opinions on the other ones - move to ``ndimage`` or just deprecate? It sounded like @WarrenWeckesser was +0.5 on deprecating in gh-6242, and I'd be okay with that as well.

**Comment by person142** [comment]
I like deprecate too. If we were to want similar functions in `ndimage`, IMO it makes more sense to rewrite them from scratch so that they don't rely on Pillow, work in nd, and have a better API (no `mode` stuff).

**Comment by rgommers** [comment]
We're not going to get to the ``misc`` cleanup for 1.0, removing the milestone for this.

**Comment by rgommers** [comment]
> We're not going to get to the ``misc`` cleanup for 1.0, removing the milestone for this.

Changing my mind, it's an awful mess. We should just get all the deprecations in.

**Comment by rgommers** [comment]
Thanks for having a go at this @MartinThoma! We're deprecating the pilutil functions instead of moving them, so I'm closing this PR.

**Pull request closed without merging**